### PR TITLE
Disconnect the network before reconnects.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/client.toit [hunk 1]
```diff
@@ -40,6 +42,36 @@ class Client:
       max_qos := 1
       subscription_callbacks_.set topic (CallbackEntry_ callback max_qos)
 
+  /**
+  Variant of $(constructor --transport) that connects to the given $host:$port over TCP.
+  */
+  constructor
+      --host/string
+      --port/int=1883
```
Comment (kasperl) [inline src/client.toit:50]: The types, =, and default values have varying spaces around them.

File: src/client.toit [hunk 2]
```diff
@@ -40,6 +42,36 @@ class Client:
...
+      --routes / Map = {:}:
+    transport := TcpTransport --host=host --port=port --net_open=net_open
+    return Client --transport=transport --logger=logger --routes=routes
+
+  /**
+  Variant of $(constructor --host) that supports TLS.
+  */
+  constructor.tls
+      --host/string
+      --port/int=8883
```
Comment (kasperl) [inline src/client.toit:62]: Also here.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/client.toit (hunk 1 above)
Comment (floitsch) [reply src/client.toit]: more consistent now.

File: src/client.toit [hunk 3]
```diff
@@ -40,6 +42,36 @@ class Client:
...
 
+  /**
+  Variant of $(constructor --transport) that connects to the given $host:$port over TCP.
+  */
+  constructor
+      --host/string
+      --port/int=1883
+      --net_open /Lambda? = (:: net.open)
+      --logger /log.Logger = log.default
+      --routes / Map = {:}:
```
[the remark this answers is not in the document]
Comment (floitsch) [reply src/client.toit]: done.

**Pull request merged**